### ISSUE TITLE
getting false positives on escaped pounds

### ISSUE THREAD

**Issue by esanfordwss** [cflint/CFLint#571]
https://github.com/cflint/CFLint/blob/ea8ffcdf83a73b5238ed02276f2463ebde44679c/src/main/java/com/cflint/plugins/core/QueryParamChecker.java#L26

Should be [^#]*#[^#] to instead of .*#[^#]

**Comment by ryaneberly** [comment]
@esanfordwss,  Thanks!

**Comment by KamasamaK** [comment]
There should be a test added that would have failed with the old pattern. Also, there should be one that handles more complex combinations like
```
<cfquery name="query" datasource="dsn">
	SELECT	*
	FROM	Table
	WHERE	column = '###foo###'
</cfquery>
```

**Comment by ryaneberly** [comment]
@KamasamaK.  Makes sense, reopening.,

**Comment by KamasamaK** [comment]
I just tested with the following in 1.4.1, and still get a false positive
```
<cfquery name="testQuery" datasource="dsn">
	SELECT	*
	FROM	Table
	WHERE	column = '##Hello##'
</cfquery>
```
